streams/src/test/java/org/apache/kafka/streams/integration/RegexSourceIntegrationTest.java

```diff
@@ -159,23 +156,26 @@ public void subscribe(final Pattern topics, final ConsumerRebalanceListener list
             }
         });
 
+        try {
+            streams.start();
+            TestUtils.waitForCondition(new TestCondition() {
+                @Override
+                public boolean conditionMet() {
+                    return assignedTopics.equals(expectedFirstAssignment);
```

Comment:
I was curious and still have Java7 installed and could reproduce the issue.

The lists are not sorted here and thus the check fails.

The reason is the following exception:
```
java.lang.UnsupportedOperationException
	at java.util.concurrent.CopyOnWriteArrayList$COWIterator.set(CopyOnWriteArrayList.java:1049)
	at java.util.Collections.sort(Collections.java:159)
	at org.apache.kafka.streams.integration.RegexSourceIntegrationTest$TheConsumerRebalanceListener.onPartitionsAssigned(RegexSourceIntegrationTest.java:445)
```

Seems `CopyOnWriteArrayList` cannot be sorted.